cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java

```diff
@@ -150,246 +154,253 @@ public void onFailure(int statusCode, String response, Throwable throwable) {
-			&& lastSessionTimestamp > 0
-			&& (System.currentTimeMillis() - lastSessionTimestamp) < MIN_SESSION_LENGTH
-		) {
-    		return;
-		}
-
-    	if (popups.size() > 0) {
-    		for (AppBannerPopup popup : popups) {
-				popup.dismiss();
-			}
-    		popups = new ArrayList<>();
-		}
-
-		lastSessionTimestamp = System.currentTimeMillis();
-
-    	sessions += 1;
-		this.saveSessions();
-
-		banners = null;
-		handler.post(this::loadBanners);
-		this.startup();
-	}
-
-	private int getSessions() {
-		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(CleverPush.context);
-		return sharedPreferences.getInt(CleverPushPreferences.APP_BANNER_SESSIONS, 0);
-	}
-
-	private void saveSessions() {
-		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(CleverPush.context);
-		SharedPreferences.Editor editor = sharedPreferences.edit();
-		editor.putInt(CleverPushPreferences.APP_BANNER_SESSIONS, sessions);
-		editor.apply();
-	}
-
-	public void triggerEvent(String key, String value) {
-		events.put(key, value);
-
-		this.startup();
-	}
-
-	public void getBanners(AppBannersListener listener) {
-		this.getBanners(listener, null);
-	}
-
-	public void getBanners(AppBannersListener listener, String notificationId) {
-		if (listener != null) {
-			if (notificationId != null) {
-				// reload banners because the banner might have been created just seconds ago
-				getBannersListeners.add(listener);
-				handler.post(() -> {
-					this.loadBanners(notificationId);
-				});
-			} else {
-				if (banners == null) {
-					getBannersListeners.add(listener);
-				} else {
-					listener.ready(banners);
-				}
-			}
-		}
-	}
+    private void sendBannerEvent(String event, Banner banner) {
+        Log.d(TAG, "sendBannerEvent: " + event);
+
+        String subscriptionId = null;
+        if (CleverPush.getInstance(activity).isSubscribed()) {
+            subscriptionId = CleverPush.getInstance(activity).getSubscriptionId();
+        }
+
+        JSONObject jsonBody = new JSONObject();
+        try {
+            jsonBody.put("bannerId", banner.getId());
+            jsonBody.put("channelId", channel);
+            jsonBody.put("subscriptionId", subscriptionId);
+        } catch (JSONException ex) {
+            Log.e("CleverPush", ex.getMessage(), ex);
+        }
+
+        CleverPushHttpClient.post("/app-banner/event/" + event, jsonBody, new CleverPushHttpClient.ResponseHandler() {
+            @Override
+            public void onSuccess(String response) {
+
+            }
+
+            @Override
+            public void onFailure(int statusCode, String response, Throwable throwable) {
+                Log.e(TAG, "App Banner Event failed." +
+                        "\nStatus code: " + statusCode +
+                        "\nResponse: " + response
+                );
+            }
+        });
+    }
+
+    public void initSession(String channel) {
+        this.channel = channel;
+
+        if (
```

Comment:
i think this looks really ugly. we shouldnt do it the same way as in javascript. put !CleverPush.getInstance(activity).isDevelopmentModeEnabled() on line with if

and ) { in line with
&& (System.currentTimeMillis() - lastSessionTimestamp) < MIN_SESSION_LENGTH